src/deployCommand.ts

```diff
@@ -57,26 +53,17 @@ export abstract class DeployCommand extends SourceCommand {
     const stashKey = Stash.getKey(this.id);
     Stash.set(stashKey, { jobid: id });
   });
-
-  // the basic sfdx flag is already making sure its of the correct length
-  public static isValidDeployId = (id: string): boolean => {
-    if (id.startsWith('0Af')) {
-      return true;
-    } else {
-      throw new SfError(messages.getMessage('invalidDeployId'), 'invalidDeployId');
-    }
-  };
   /**
    * Request a report of an in-progress or completed deployment.
    *
    * @param id the Deploy ID of a deployment request
    * @returns DeployResult
    */
-  protected async report(id?: string): Promise<DeployResult> {
+  protected async report(conn: Connection, id?: string): Promise<DeployResult> {
     const deployId = this.resolveDeployId(id);
     this.displayDeployId(deployId);
 
-    const res = await this.org.getConnection().metadata.checkDeployStatus(deployId, true);
+    const res = await conn.metadata.checkDeployStatus(deployId, true);
 
     const deployStatus = res as unknown as MetadataApiDeployStatus;
```

Comment:
eww.  Is this just a mismatch between SDR's types and jsforce's ?  

What if we (probably in SDR) had a simple function to convert those mismatched arrays?